scripts/openwebnet-discovery.nse

```diff
@@ -137,10 +137,15 @@ local function get_response(sd, request)
 
     if status == nil then
       stdnse.debug("Error: " .. tostring(data))
-      -- Captures the NACK after TIMEOUT occured.
-      -- Avoids false results.
-      status, data = sd:receive_buf("##", true)
-      break
+      if tostring(data) == "TIMEOUT" then
```

Comment:
If we receive `nil` as an error, then the API we are writing to is broken and we **should** crash. Errors due to network input should be handled quietly; errors due to improper/incomplete programming should be reported loudly so they can be fixed.